metrics/tag.go

```diff
@@ -151,20 +146,28 @@ func NewTags(t map[string]string) (Tags, error) {
 	return tags, nil
 }
 
-var validChars = make(map[rune]struct{})
+var validKeyChars = make(map[rune]struct{})
```

Comment:
It seems like you can make this faster by using arrays and avoid the map access -- on my laptop this removed around 20% from the benchmark: https://github.com/palantir/pkg/compare/dt/new-tag-optomize...bm/dt/new-tag-optomize